tests/test_app.py

```diff
@@ -257,6 +257,24 @@ def test_session_stats_create_with_negative_values(self):
         sessions = Session.query.all()
         self.assertEqual(0, len(sessions))
 
+    def test_session_stats_with_provider_id(self):
+        payload = {
+            'bytes_sent': 20,
+            'bytes_received': 40,
+            'provider_id': '0x1',
+        }
+        auth = generate_test_authorization(json.dumps(payload))
+        re = self._post(
+            '/v1/sessions/123/stats',
+            payload,
+            headers=auth['headers'],
+        )
+
+        self.assertEqual(200, re.status_code)
+
+        session = Session.query.get('123')
+        self.assertEqual('0x1', session.node_key)
```

Comment:
It might be easier to implement this sending in `node` first - than we wouldn't need a separate test `with_provider_id`, because `provider_id` could be required parameter, not optional.